### PR TITLE
modify address to LowerCase

### PR DESCRIPTION
**Pull request by zzlune**
**What kind of change does this PR introduce?**

**Did you add tests for your changes?**

**Does this PR introduce a breaking change?**

**What needs to be documented once your changes are merged?**

---
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-scan/129)

### CODE REVIEW

**Review by 21paradox — DISMISSED** [review]
> Reviewed 1 of 4 files at r1, 4 of 4 files at r2.
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-scan/129)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @angelia-yuqi-personal)

**Review by 21paradox — DISMISSED** [review]
> Reviewed 6 of 6 files at r3.
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-scan/129)* status: :shipit: complete! all files reviewed, all discussions resolved (waiting on @angelia-yuqi-personal)

File: src/pages/Block/detail.js [hunk 1]
```diff
@@ -297,6 +297,10 @@ class Detail extends Component {
     const {
       match: { params },
     } = this.props;
+    this.getBlockHash = () => {
+      const { blockhash } = params;
```
Comment (AngeliaQ) [inline src/pages/Block/detail.js]: params 没定义

File: src/pages/Contract/update.js [hunk 2]
```diff
@@ -477,19 +484,17 @@ class ContractUpdate extends Component {
...
   }
 
   componentDidUpdate(prevProps) {
     const {
       match: { params },
     } = this.props;
-    const { address } = params;
-    if (address !== prevProps.match.params.address) {
-      this.fetchContactInfo(params.address);
+    const address = tranferToLowerCase(params.address);
```
Comment (AngeliaQ) [inline src/pages/Contract/update.js:494]: 这里可以直接删掉上面的params获取，直接用this.getAddress()

File: src/pages/Contract/update.js [hunk 3]
```diff
@@ -716,7 +722,7 @@ class ContractUpdate extends Component {
         <Wrapper>
           <HeadBar>
             <h1>{i18n('app.pages.contract.edit')}</h1>
-            <p>{params.address}</p>
```
Comment (AngeliaQ) [inline src/pages/Contract/update.js:719]: 上面定义params的代码可以去掉了？

File: src/pages/Block/detail.js [hunk 4]
```diff
@@ -314,17 +318,18 @@ class Detail extends Component {
...
-    this.fetchBlockDetail(params.blockhash, { activePage: 1 });
-    this.fetchReffereBlock(params.blockhash, { refBlockCurPage: 1 });
+    this.fetchBlockDetail(this.getBlockHash(), { activePage: 1 });
+    this.fetchReffereBlock(this.getBlockHash(), { refBlockCurPage: 1 });
   }
 
   componentDidUpdate(prevProps) {
     // eslint-disable-next-line react/destructuring-assignment
-    const { blockhash } = this.props.match.params;
+    const blockhash = tranferToLowerCase(this.props.match.params.blockHash);
```
Comment (AngeliaQ) [inline src/pages/Block/detail.js]: 这里可直接用this.getBlockHash()

File: src/pages/Block/detail.js [hunk 5]
```diff
@@ -297,6 +297,10 @@ class Detail extends Component {
     const {
       match: { params },
```
Comment (AngeliaQ) [inline src/pages/Block/detail.js]: 这里应该放在下面的getBlockHash函数里

**Review by AngeliaQ — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]

**Review by AngeliaQ — COMMENTED** [review]
> Reviewed 4 of 6 files at r3, 2 of 2 files at r4.
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-scan/129)* status: all files reviewed, 4 unresolved discussions (waiting on @zctocm)

**Review by AngeliaQ — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]

**Review by AngeliaQ — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/conflux-chain/conflux-scan/129)* status: :shipit: complete! all files reviewed, all discussions resolved

___
*[src/pages/Block/detail.js, line 298 at r3](https://reviewable.io/reviews/conflux-chain/conflux-scan/129#-M5aVGY0N2sbS1eJxXVL-r3-298:-M5alD3L-J3wkCUe9phk:bb7f2mf) ([raw file](https://github.com/conflux-chain/conflux-scan/blob/cd2c89ab2c67f729395f61f2f6efec5bc7666838/src/pages/Block/detail.js#L298)):*
<details><summary><i>Previously, angelia-yuqi-personal (Angelia) wrote…</i></summary><blockquote>

这里应该放在下面的getBlockHash函数里
</blockquote></details>

done
___
*[src/pages/Block/detail.js, line 301 at r3](https://reviewable.io/reviews/conflux-chain/conflux-scan/129#-M5aVGY0N2sbS1eJxXVL-r3-301:-M5al4x9ADvM7tjTxcjn:b-7rl0kd) ([raw file](https://github.com/conflux-chain/conflux-scan/blob/cd2c89ab2c67f729395f61f2f6efec5bc7666838/src/pages/Block/detail.js#L301)):*
<details><summary><i>Previously, angelia-yuqi-personal (Angelia) wrote…</i></summary><blockquote>

params 没定义
</blockquote></details>

done
___
*[src/pages/Block/detail.js, line 327 at r3](https://reviewable.io/reviews/conflux-chain/conflux-scan/129#-M5aVGY0N2sbS1eJxXVL-r3-327:-M5alEdE-tIeHpMT1qFb:bb7f2mf) ([raw file](https://github.com/conflux-chain/conflux-scan/blob/cd2c89ab2c67f729395f61f2f6efec5bc7666838/src/pages/Block/detail.js#L327)):*
<details><summary><i>Previously, angelia-yuqi-personal (Angelia) wrote…</i></summary><blockquote>

这里可直接用this.getBlockHash()
</blockquote></details>

done
___
*[src/pages/Contract/update.js, line 494 at r3](https://reviewable.io/reviews/conflux-chain/conflux-scan/129#-M5aO3TAgDi3C8obRbFq-r3-494:-M5alG6u7k2qdxem7b_x:bb7f2mf) ([raw file](https://github.com/conflux-chain/conflux-scan/blob/cd2c89ab2c67f729395f61f2f6efec5bc7666838/src/pages/Contract/update.js#L494)):*
<details><summary><i>Previously, angelia-yuqi-personal (Angelia) wrote…</i></summary><blockquote>

这里可以直接删掉上面的params获取，直接用this.getAddress()
</blockquote></details>

done

**Pull request merged**